src/character_attire.cpp

```diff
@@ -2245,52 +2300,71 @@ void outfit::add_stash( Character &guy, const item &newit, int &remaining_charge
-        std::list<item *> items;
-        item_location carried_item = guy.get_wielded_item();
-        if( carried_item != item_location::nowhere ) {
-            items.emplace_back( &*carried_item );
-        }
-        for( item &i : worn ) {
-            items.emplace_back( &i );
-        }
-        for( item *i : items ) {
-            for( const item_pocket *pocket : i->get_all_contained_pockets() ) {
-                if( pocket->can_contain( temp_it ).success() ) {
-                    // Top-level( wielded, worn ) pockets may be stored
-                    found_pocket = const_cast<item_pocket *>( pocket );
-                    pockets.emplace_back( found_pocket );
-                }
-                for( const item *contained : pocket->all_items_ptr( item_pocket::pocket_type::CONTAINER ) ) {
-                    for( const item_pocket *pocket_nest : contained->get_all_contained_pockets() ) {
-                        if( pocket_nest->can_contain( temp_it ).success() && pocket_nest->rigid() ) {
-                            // Nested pocket with rigid() can only be stored
-                            found_pocket = const_cast<item_pocket *>( pocket_nest );
-                            pockets.emplace_back( found_pocket );
-                        }
-                    }
-                }
+        std::vector<pocket_data_with_parent> pockets_with_parent;
+        std::vector<pocket_data_with_parent> pockets_with_parent_working;
+        pockets_with_parent_working = get_all_pocket_with_parent( guy );
+
+        for( const pocket_data_with_parent &pocket_data : pockets_with_parent_working ) {
+            const item_pocket *pocket = pocket_data.pocket_ptr;
+            if( pocket->can_contain( temp_it ).success() ) {
+                pockets_with_parent.emplace_back( pocket_data );
             }
         }
 
-        // Sort by item_pocket::better_pocket
-        std::sort( pockets.begin(), pockets.end(), [newit, temp_it]( item_pocket * lhs,
-        item_pocket * rhs ) {
-            return rhs->better_pocket( *lhs, newit, false );
+        std::sort( pockets_with_parent.begin(), pockets_with_parent.end(), [newit, temp_it]
+        ( const pocket_data_with_parent & lhs, const pocket_data_with_parent & rhs ) {
+            const item_pocket *lpd = lhs.pocket_ptr;
+            const item_pocket *rpd = rhs.pocket_ptr;
+            const item_pocket &lpp = *lpd;
+            const int lp_nested_lvl = lhs.nested_level;
+            const int rp_nested_lvl = rhs.nested_level;
+
+            if( lpd->settings.priority() != rpd->settings.priority() ) {
+                return lpd->settings.priority() > rpd->settings.priority();
+            }
+            if( lp_nested_lvl != rp_nested_lvl ) {
+                return lp_nested_lvl < rp_nested_lvl;
+            }
+            if( lpd->rigid() != rpd->rigid() ) {
+                return lpd->rigid();
+            }
+            return rpd->better_pocket( lpp, temp_it, false );
```

Comment:
I assume this is all done just for the nesting level preference? While that is certainly nice to have, I don't think it warrants bypassing `better_pocket` like this, because whenever that gets touched this would need to be checked as well. You either need to work it into `better_pocket` somehow or ignore it.

As an aside, `better_pocket` currently has a slight bug with it's fallback `obtain_cost` check. It currently can't check the true `obtain_cost` since it isn't aware of any parent pockets. If that was properly checked, there would be no need to worry about the nesting level at all, because deeper nested pockets are naturally slower to access and would lose in a comparison. But that's definitely not something to fix in this PR.

Reply:
> I don't think it warrants bypassing better_pocket like this, because whenever that gets touched this would need to be checked as well. You either need to work it into better_pocket somehow or ignore it.

I understand. My mind was also a bit confused.
I used to think that the nesting level should be incorporated into the sorting priority, but now I think that is more of a game design issue than a programming issue.
After this, I would like to brush up on the overall process and then test it, and if either way ( discard the unique condition and use only better_pocket or not use better_pocket) works as expected, then I would like to do so.

Reply:
> discard the unique condition and use only better_pocket or not use better_pocket

You absolutely have to use `better_pocket`, because we can't have two different implementations of (automatically) selecting a pocket to insert items into. The question is just if `better_pocket` is missing something that you want to implement in the scope of this PR.

Reply:
I can understand what you are saying. I would like to adopt better_pocket this time.

Reply:
Changed sorting conditions to be delegated to better_pocket.